### PR TITLE
Require user be present when Creating or changing an idea

### PR DESCRIPTION
**Pull request by MikeRogers0**
This adds the requirement that only Authenticated user can post ideas & only they can update their own ideas.

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -39,6 +39,7 @@ end
 
 group :test do
   gem 'capybara', '>= 3.26'
+  gem 'faker', require: false
```
Comment (MikeRogers0) [inline Gemfile:42]: Faker - Awesome gem for setting up some made data in our tests.

File: app/controllers/ideas_controller.rb [hunk 2]
```diff
@@ -54,4 +56,10 @@ def set_idea
   def idea_params
     params.require(:idea).permit(:title, :description)
   end
+
+  def authenticate_user_can_change_idea!
+    return if @idea.user == current_user
+
+    redirect_to ideas_url, notice: "You can't update that Idea"
+  end
```
Comment (MikeRogers0) [inline app/controllers/ideas_controller.rb:64]: Normally I'd use something like Pundit to achieve this, but this looked easier to explain :)

File: spec/support/devise.rb [hunk 3]
```diff
@@ -0,0 +1,5 @@
+RSpec.configure do |config|
+  config.include Devise::Test::ControllerHelpers, type: :controller
+  config.include Devise::Test::IntegrationHelpers, type: :request
+  config.include Devise::Test::IntegrationHelpers, type: :feature
+end
```
Comment (MikeRogers0) [inline spec/support/devise.rb:5]: This lets us use `sign_in` in our tests to login users :)

**Review by meg-gutshall — APPROVED** [review]

**Comment by meg-gutshall** [comment]
Yay! Faker!

**Pull request merged**